src/librustdoc/clean/utils.rs

```diff
@@ -73,31 +80,128 @@ pub(crate) fn krate(cx: &mut DocContext<'_>) -> Crate {
+                        params[index].def_id,
+                    ) {
+                        return None;
+                    }
+
+                    elision_has_failed_once_before = true;
+                }
+
+                Some(GenericArg::Type(clean_middle_ty(substs.rebind(ty), cx, None)))
+            }
+            GenericArgKind::Const(ct) => {
+                if !elision_has_failed_once_before
+                    && let Some(params) = params
+                    && let Some(default) = params[index].default_value(cx.tcx)
+                {
+                    let default =
+                        substs.map_bound(|substs| default.subst(cx.tcx, substs).expect_const());
+
+                    if can_elide_generic_arg(
+                        cx.tcx,
+                        &cause,
+                        param_env,
+                        substs.rebind(ct),
+                        default,
+                        params[index].def_id,
+                    ) {
+                        return None;
+                    }
+
+                    elision_has_failed_once_before = true;
+                }
+
+                Some(GenericArg::Const(Box::new(clean_middle_const(substs.rebind(ct), cx))))
+            }
         }
-    }));
-    ret_val
+    };
+
+    // FIXME: Yikes, there gotta exist a better & faster solution!!
+    //        Maybe partition the substs by `has_default` and only reverse-iterate for those
+    //        that may have defaults somehow?
+    args.extend(substs.skip_binder().iter().enumerate().rev().filter_map(subst_to_arg));
+    args.reverse();
+    args
+}
+
+fn can_elide_generic_arg<'tcx, T: ToTrace<'tcx> + TypeVisitable<TyCtxt<'tcx>>>(
+    tcx: TyCtxt<'tcx>,
+    cause: &ObligationCause<'tcx>,
+    param_env: ty::ParamEnv<'tcx>,
+    actual: ty::Binder<'tcx, T>,
+    default: ty::Binder<'tcx, T>,
+    did: DefId,
+) -> bool {
+    assert!(!actual.has_infer());
+    assert!(!default.has_infer());
+
+    // We intentionally look for *arbitrary* late-bound regions instead of only *escaping* ones as we
+    // want to identify non-escaping alpha-equivalent late-bound regions later.
+    if !actual.has_late_bound_regions()
+        && !actual.has_projections()
+        && !default.has_late_bound_regions()
+        && !default.has_projections()
+    {
+        // fast pass
+        return actual.skip_binder() == default.skip_binder();
+    }
+
+    let infcx = tcx.infer_ctxt().build();
+    let ocx = ObligationCtxt::new(&infcx);
+
+    // FIXME: This is obviously incorrect. Not sure what one is supposed to do in this situation though.
+    let aggressively_liberate = |arg: ty::Binder<'tcx, T>| {
+        tcx.fold_regions(arg.skip_binder(), |r, _| match *r {
+            ty::ReLateBound(_, br) => ty::Region::new_free(tcx, did, br.kind),
+            _ => r,
+        })
+    };
+    let actual = aggressively_liberate(actual);
+    let default = aggressively_liberate(default);
```

Comment:
```suggestion
    def aggressively_liberate(arg: ty::Binder<'tcx, T>, tcx: ...) -> ...{
        tcx.fold_regions(arg.skip_binder(), |r, _| match *r {
            ty::ReLateBound(_, br) => ty::Region::new_free(tcx, did, br.kind),
            _ => r,
        })
    };
    let actual = aggressively_liberate(actual, tcx);
    let default = aggressively_liberate(default, tcx);
```

Same as above (you could do similar with that inside closure)